src/Analysis/GWASWizard/shared/AttritionTable.jsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Collapse, List, Table } from 'antd';
+import AttritionTableRow from './AttritionTableRow';
+import '../../GWASUIApp/GWASUIApp.css';
+import { attritionTableHeaderConfig, headerDataSource } from './constants';
+
+const { Panel } = Collapse;
+
+const AttritionTable = ({
+  selectedCohort,
+  otherSelectedCohort,
+  outcome,
+  selectedCovariates,
+  selectedDichotomousCovariates,
+  sourceId,
+  tableHeader,
+}) => {
+  const [covariateSubsets, setCovariateSubsets] = useState([]);
+  const getCovariateRow = (selectedCovs = [], selectedCustomdichotomousCovs = []) => {
+    const subsets = [];
+    // todo: handle case of deselecting/selecting existing covariates (100% missing?) after adding custom dichotomous
```

Comment:
a clearer name for this function would also be good (maybe `getCovariateRows` to reflect the fact that it is returning items for multiple rows?) and perhaps some comments explaining what it is meant to return when given `(selectedCovs = [], selectedCustomdichotomousCovs = [])`, as it is not a very obvious/common result.

Reply:
its being called for each row so i made it singular for that reason. also i changed those names because linter didn't like the overshadowing of variable names

Reply:
gotta address an edge case here. can include this as part of that ticket